sympy/combinatorics/permutations.py

```diff
@@ -983,7 +983,7 @@ def __new__(cls, *args, size=None, **kwargs):
         temp = set(temp)
 
         if not is_cycle:
-            if any(i not in temp for i in range(len(temp))):
```

Comment:
Not sure if this actually helps performance, but tends to be more readable in my opinion. It may avoid multiple not, but not sure if `not in` is one or two statements.

Reply:
Since `temp` is a set here this can be something like:
```python
if temp != set(range(len(temp))):
```